cmd/download/rosa/cmd.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rosa
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/dustin/go-humanize"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/cmd/verify/rosa"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "rosa-client",
+	Aliases: []string{"rosa"},
+	Short:   "Download ROSA client tools",
+	Long:    "Downloads to latest compatible version of the ROSA client tools.",
+	Example: `  # Download rosa client tools
+  rosa download rosa`,
+	Run: run,
+}
+
+func run(cmd *cobra.Command, argv []string) {
+	reporter := rprtr.CreateReporterOrExit()
+
+	platform := getPlatform()
+	extension := getExtension()
+
+	filename := fmt.Sprintf("rosa-%s.%s", platform, extension)
+
+	downloadURL := fmt.Sprintf("%s/%s", rosa.DownloadLatestMirrorFolder, filename)
+
+	reporter.Infof("Downloading %s to your current directory", downloadURL)
+
+	err := download(downloadURL, filename)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	reporter.Infof("Successfully downloaded %s", filename)
+}
+
+// Get the platform name used on the oc tarball filename
+func getPlatform() string {
+	if runtime.GOOS == "darwin" {
+		return "macosx"
+	}
+	return runtime.GOOS
+}
+
+// Get the extension used for the compressed oc file
+func getExtension() string {
+	if runtime.GOOS == "windows" {
+		return "zip"
+	}
+	return "tar.gz"
+}
+
+// download will download a url to a local file. It's efficient because it will
+// write as it downloads and not load the whole file into memory. We pass an io.TeeReader
+// into Copy() to report progress on the download.
+func download(url string, filename string) error {
```

Comment:
What is the difference between this function and the function in `cmd/download/oc/cmd.go`, can we re-use the function?